### PR TITLE
Add documentation on public test suites

### PR DESCRIPTION
**Pull request by meghan-kradolfer**
_**NOTE: This should not be merged until the feature flag `ANALYTICS_PUBLIC_SUITE_ACCESS` is turned on for all users**_

We are about to release a new feature that introduces public test suites into Test Analytics. We need to make sure users understand what public suites are so it is used appropriately. There is a risk that users will make their suite public without realising the privacy implications if it is not documented.

I have copied what is existing in the docs for [Public pipelines](https://buildkite.com/docs/pipelines/public-pipelines#make-a-pipeline-public-using-the-ui) and altered the content to relate to test suites

[Linear ticket](https://linear.app/buildkite/issue/PIE-2168/add-documentation-on-public-test-suites-to-the-docs)

---

Screenshot 2023-11-29 at 4 25 00 PM

### CODE REVIEW

**Review by KatieWright26 — COMMENTED** [review]
I reckon it could be worth expanding on how to turn on public suites, as only org admins will have this power initially, until the org TA settings are updated to grant this permission to all users. Maybe something like:

"Permission to make a test suite public is reserved for organization admins initially. Admins are able to extend this permission to all organization members via the Security tab in the organization settings."

Screenshot 2023-11-29 at 11 00 48

File: pages/test_analytics/public_test_suites.md
Comment (KatieWright26) [inline pages/test_analytics/public_test_suites.md:10]: This might read as though the test code is public, when actually it's the analytics?

File: pages/test_analytics/public_test_suites.md
Comment (KatieWright26) [inline pages/test_analytics/public_test_suites.md:13]: I'm pretty sure "span data" is a buildkite concept, and it doesn't apply to all users so could be good to be more specific here. Maybe something closer to:
```suggestion
- Test execution data. For those using buildkites ruby test collector this includes SQL query data, HTTP request paths and the execution timeline for test-collector-ruby users 
- Run environmental variables
```

Comment (meghan-kradolfer) [reply pages/test_analytics/public_test_suites.md:13]: I pulled this from our confirmation popup to make suites public, would we also need to update that there?
Screenshot 2023-11-29 at 11 25 34 AM

Comment (KatieWright26) [reply pages/test_analytics/public_test_suites.md:13]: Hm, not sure? Keen for @jasmine-q's take, as the variables relate to a run, not a suite per se.

File: pages/test_analytics/public_test_suites.md
Comment (KatieWright26) [inline pages/test_analytics/public_test_suites.md:5]:
```suggestion
```

**Review by KatieWright26 — APPROVED** [review]
LGTM; would be good to get a review from @mbelton-buildkite before merging

File: pages/test_analytics/public_test_suites.md
Comment (mbelton-buildkite) [inline pages/test_analytics/public_test_suites.md:5]: General nitpicks:

```suggestion
Making a suite public gives read-only access to all users. This means users who are unauthenticated or belong to another organization can view the following:
```

File: pages/test_analytics/public_test_suites.md
Comment (mbelton-buildkite) [inline pages/test_analytics/public_test_suites.md:11]: Capitals and commas

```suggestion
- Test execution data. For those using Buildkite's Ruby test collector, this includes SQL query data, HTTP request paths, and the execution timeline.
```

File: pages/test_analytics/public_test_suites.md
Comment (mbelton-buildkite) [inline pages/test_analytics/public_test_suites.md:13]: Let's give each one a bullet point:

```suggestion
  + `commit_sha`
  + `branch`
  + `message`
  + `url`
  + `number`
  + `job_id`
```

File: pages/test_analytics/public_test_suites.md
Comment (mbelton-buildkite) [inline pages/test_analytics/public_test_suites.md:19]: Make active:

```suggestion
You make a test suite public from the suite's _Settings_:
```

File: pages/test_analytics/public_test_suites.md
Comment (mbelton-buildkite) [inline pages/test_analytics/public_test_suites.md:23]: Style nitpicks:

```suggestion
Only organization admins have permission to make a test suite public by default. Admins can extend this permission to all organization members from the _Security_ tab in the organization settings.
```

File: pages/test_analytics/public_test_suites.md
Comment (mbelton-buildkite) [inline pages/test_analytics/public_test_suites.md:15]: Style nitpick: Just use simple sentences here.

```suggestion
Before making a suite public, you should verify that runs do not expose sensitive information in their logs or environment variables. This applies to both new and historical runs.
```

**Review by mbelton-buildkite — APPROVED** [review]
I've left a bunch of suggestions, but in general, it looks good to me. 😊

**Pull request merged**
